ext/spl/spl_iterators.stub.php

```diff
@@ -0,0 +1,475 @@
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+}
+
+class AppendIterator extends IteratorIterator
+{
+    public function __construct() {}
+
+    /** @return void */
+    public function append(Iterator $iterator) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+
+    /** @return int */
+    public function getIteratorIndex() {}
+
+    /** @return ArrayIterator */
+    public function getArrayIterator() {}
+}
+
+class InfiniteIterator extends IteratorIterator
+{
+    public function __construct(Iterator $iterator) {}
+
+    /** @return void */
+    public function next() {}
+}
+
+class RegexIterator extends FilterIterator
+{
+    public function __construct(Iterator $iterator, string $regex, int $mode = self::MATCH, int $flags = 0, int $preg_flags = 0) {}
+
+    /** @return bool */
+    public function accept() {}
+
+    /** @return int */
+    public function getMode() {}
+
+    /** @return void */
+    public function setMode(int $mode) {}
+
+    /** @return int */
+    public function getFlags() {}
+
+    /** @return void */
+    public function setFlags(int $flags) {}
+
+    /** @return string */
+    public function getRegex() {}
+
+    /** @return int */
+    public function getPregFlags() {}
+
+    /** @return void */
+    public function setPregFlags(int $preg_flags) {}
+}
+
+class RecursiveRegexIterator extends RegexIterator implements RecursiveIterator
+{
+    public function __construct(RecursiveIterator $iterator, string $regex, int $mode = self::MATCH, int $flags = 0, int $preg_flags = 0) {}
+
+    /** @return bool */
+    public function accept() {}
+
+    /** @return bool */
+    public function hasChildren() {}
+
+    /** @return RecursiveRegexIterator */
+    public function getChildren() {}
+}
+
+class RecursiveTreeIterator extends RecursiveIteratorIterator
+{
+    /**
+     * @param RecursiveIterator|IteratorAggregate $iterator
+     * @param int $caching_it_flags
+     */
+    public function __construct($iterator, int $flags = self::BYPASS_KEY, $caching_it_flags = CachingIterator::CATCH_GET_CHILD,
+                                int $mode = self::SELF_FIRST) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return string */
+    public function key() {}
+
+    /** @return string|null */
+    public function current() {}
```

Comment:
Looks like this can return non-string for RTIT_BYPAS_CURRENT?

Reply:
Unfortunately, I'm not able to determine :/ But it seems like -> marking it as `mixed`